examples/snake.rs

```diff
@@ -0,0 +1,324 @@
+    PlayAgain,
+}
+impl Game for SnakeGame {
+    const TICKS_PER_SECOND: u16 = 60;
+    type Input = CustomInput;
+    type LoadingScreen = ();
+
+    fn load(_window: &Window) -> Task<SnakeGame> {
+        let mut snake = Snake::new();
+        snake.create_snake();
+        Task::succeed(|| SnakeGame {
+            snake,
+            score: 0,
+            apple: Apple::new(),
+            ticks: 0,
+            speed: 20,
+            last_key: None,
+        })
+    }
+    fn update(&mut self, _window: &Window) {}
+    fn interact(&mut self, input: &mut CustomInput, _window: &mut Window) {
+        if input.keys_pressed.len() != 0 {
+            let key = input.keys_pressed[0];
+            match key {
+                KeyCode::Right => {
+                    self.last_key = Some(key);
+                }
+                KeyCode::Left => {
+                    self.last_key = Some(key);
+                }
+                KeyCode::Down => {
+                    self.last_key = Some(key);
+                }
+                KeyCode::Up => {
+                    self.last_key = Some(key);
+                }
+                _ => (),
+            }
+        }
+    }
+    fn draw(&mut self, frame: &mut Frame, timer: &Timer) {
+        if timer.has_ticked() && !self.snake.ate_himself() {
+            self.ticks += 1;
+            if self.ticks == self.speed {
+                // We don't want our snake to move 60 times per seconds.
+                self.snake.move_to(self.last_key);
+                if self.snake.ate_apple(self.apple.get_pos()) {
+                    self.apple.eaten = true;
+                    self.score += 1;
+                    self.snake.add_queue();
+                    if self.score % 2 == 0 {
+                        self.speed -= 1; // Every two apples, the snake go faster !
+                    }
+                }
+                self.ticks = 0;
+            }
+        }
+        if !self.snake.ate_himself() {
+            frame.clear(Color::new(0.5, 0.5, 0.5, 1.0));
+            self.snake.draw_snake(frame);
+            self.apple.draw(frame);
+        } else {
+            frame.clear(Color::from_rgb(255, 0, 0));
+            let mut font = Font::from_bytes(
+                frame.gpu(),
+                include_bytes!("../resources/font/Inconsolata-Regular.ttf"),
+            )
+            .expect("Failed to load font");
+
+            font.add(Text {
+                content: "You lost",
+                position: Point::new(330.0, 300.0),
+                size: 50.0,
+                color: Color::WHITE,
+                ..Text::default()
+            });
+            font.add(Text {
+                content: format!("Your score is: {}", self.score).as_str(),
+                position: Point::new(270.0, 200.0),
+                size: 50.0,
+                color: Color::WHITE,
+                ..Text::default()
+            });
+            font.draw(&mut frame.as_target());
+        }
```

Comment:
We could use [`coffee::ui`](https://docs.rs/coffee/0.3.2/coffee/ui/index.html) to draw the game over screen and maybe the current score during gameplay.